syntaxes/template.json

```diff
@@ -31,25 +31,41 @@
       ]
     },
 
-    "propertyBinding": {
-      "begin": "(\\[.*\\])(=)(\"|')",
+    "ngBinding": {
+      "begin": "(([\\[\\(]{1,2})(?:\\s*)(@?[-_a-zA-Z0-9.$]*)(?:\\s*)([\\]\\)]{1,2}))(=)([\"'])",
```

Comment:
I would rather keep three separate rules for property, event, and two-way bindings, because a merged pattern will run into issues of matching mixed patterns, like `[prop)`.

Reply:
@ayazhafiz Do you think I should go ahead and split them up in this PR, or keep this PR to just property binding and add event/two-way binding in future PRs? I don't mind doing the work here if you wouldn't object to a bigger diff

Reply:
Let's keep this one to property binding and add the rest in separate PRs ))

Reply:
@ayazhafiz This should be taken care of now